### PR TITLE
DEV-1328 Check if md5 is valid

### PR DESCRIPTION
**Pull request by RudolfDG**

### CODE REVIEW

File: main.py [hunk 1]
```diff
@@ -183,10 +183,14 @@ def handle_create_event(event: dict, properties, ctx: Context) -> bool:
     """Handler for s3 create events"""
     # Check if item already in mediahaven based on key and md5
     mediahaven_service = MediahavenService(ctx)
-    query_params = [
-        ("s3_object_key", f'"{get_from_event(event, "object_key")}"'),
-        ("md5", get_from_event(event, "md5")),
-    ]
+    try:
+        query_params = [
+            ("s3_object_key", f'"{get_from_event(event, "object_key")}"'),
```
Comment (spacid) [inline main.py]: Can't this also result in a KeyError?

File: meemoo/helpers.py [hunk 2]
```diff
@@ -35,14 +36,21 @@ def try_to_find_md5(object_metadata):
...
         if key in object_metadata.keys():
             log.debug(f"md5sum located in metadata-field: '{key}'")
-            return object_metadata[key]
-    return ""
+            md5 = object_metadata[key]
+    
+    if re.match("^[a-fA-F0-9]{32}$", md5):
+        return md5
+    else:
+        raise KeyError(f"md5 not found or semantically incorrect. Found: '{md5}'")
```
Comment (spacid) [inline meemoo/helpers.py]: A KeyError is typically for when a key is not found in a dictionary. Maybe a custom error would be more clear?

File: meemoo/helpers.py [hunk 3]
```diff
@@ -50,12 +58,11 @@ def try_to_find_md5(object_metadata):
     if re.match("^[a-fA-F0-9]{32}$", md5):
         return md5
     else:
-        raise KeyError(f"md5 not found or semantically incorrect. Found: '{md5}'")
+        raise InvalidEventException(f"md5 not found or semantically incorrect. Found: '{md5}'")
 
 
 def get_from_event(event, name):
```
Comment (spacid) [inline meemoo/helpers.py:64]: It would be better if the event gets parsed once, into an instance of an event class. Out of scope for this ticket.

**Review by spacid — APPROVED** [review]

File: meemoo/helpers.py [hunk 4]
```diff
@@ -27,27 +28,41 @@
...
 def try_to_find_md5(object_metadata):
     """Simple convenience function that allows to be able to try different
     possible md5 field names (keys) and return it's value.
     Args:
         object_metadata (dict): The object metadata sub-section of the S3-event.
     Returns:
-        str: The md5sum when found. An empty string ('') otherwise.
+        str: The md5sum when found.
+    Raises:
+        KeyError: The md5sum was not found or is semantically incorrect.
```
Comment (maartends) [inline meemoo/helpers.py]:
```suggestion
        KeyError: The md5sum was not found or is syntactically incorrect.
```

File: meemoo/helpers.py [hunk 5]
```diff
@@ -27,27 +28,41 @@
...
         if key in object_metadata.keys():
             log.debug(f"md5sum located in metadata-field: '{key}'")
-            return object_metadata[key]
-    return ""
+            md5 = object_metadata[key]
+    
+    if re.match("^[a-fA-F0-9]{32}$", md5):
+        return md5
+    else:
+        raise InvalidEventException(f"md5 not found or semantically incorrect. Found: '{md5}'")
```
Comment (maartends) [inline meemoo/helpers.py]:
```suggestion
        raise InvalidEventException(f"md5 not found or syntactically incorrect. Found: '{md5}'")
```

**Review by maartends — APPROVED** [review]
Tests (positive and negative) should also be added for:
- `is_event_valid`
- `try_to_find_md5`.

(new ticket)

**Review by maartends — APPROVED** [review]
:+1:

**Pull request merged**